### PR TITLE
Add action to support docs merge safety

### PR DESCRIPTION
**Pull request by jhgilbert**
This action sends the PR title, URL, and author to the provided Slack webhook if:

- The actor is not a member of the provided team
- The PR is already approved by a member of the provided team

I'm adding this on behalf of the docs team. Our use case:

We want to improve merge safety in the documentation repo. PR authors often want us to wait on merging PRs, but forget to communicate that to us. Similarly, while we provide a "okay to merge" checkbox for them in the PR description, they often don't see the checkbox, so the oncall has to reach out to them and verify that they want the PR merged.

To add clarity, we're planning to add "Do Not Merge" to new PRs in the documentation repo by default, and post a comment to the author explaining that we will still review the PR unless they mark it as WIP, we just won't ship it until they remove "Do Not Merge". [This test PR](https://github.com/DataDog/jen.gilbert-actions-sandbox/pull/60) gives you an idea of how that looks.

By the time the author removes the "Do Not Merge" label, someone on the docs team may have already approved the PR. In that case, we'd like to be notified, so we can merge the PR without the author having to come ask the oncall to do it.

This action, which we'll run when "Do Not Merge" is removed, notifies the provided Slack webhook if certain conditions are met: if the actor (in our use case, the label remover) is NOT a team member, and if the PR is already approved by a team member. Otherwise, the action exits without doing anything.

We're using the action to provide this message in Slack:
Screenshot 2024-08-20 at 3 49 10 PM

I haven't written many GitHub actions/workflows, so any and all feedback is welcome. I'll also seek out a security review for it before merging it. Thank you!

### CODE REVIEW

File: src/external-actor-on-team-approved-pr.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import * as core from "@actions/core";
+import * as httpm from "@actions/http-client";
```
Comment (skarimo) [inline src/external-actor-on-team-approved-pr.ts:2]: Where is this dependency coming from? Looks like we are missing it in package.json

File: src/external-actor-on-team-approved-pr.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import { context, getOctokit } from "@actions/github";
+
+process.on("unhandledRejection", handleError);
+main().catch(handleError);
+
+async function main(): Promise<any> {
+  const http: httpm.HttpClient = new httpm.HttpClient(
+    "external-actor-on-team-approved-pr"
```
Comment (skarimo) [inline src/external-actor-on-team-approved-pr.ts:10]: nit: Would be nice if the user agent is a bit more specific:
`actions/external-actor-on-team-approved-pr`

File: external-actor-on-team-approved-pr/action.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: Merge readiness notification
+description: Send a Slack notification when a team-reviewed PR is ready to merge.
+inputs:
+  api-token:
```
Comment (skarimo) [inline external-actor-on-team-approved-pr/action.yml:4]:
```suggestion
  github-token:
```

File: src/external-actor-on-team-approved-pr.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      }
+    }
+  }
+
+  // If no one on the team has approved it yet,
+  // skip the Slack notification
+  if (!isApproved) {
+    console.log(
+      "No team members have approved the PR yet, skipping Slack notification."
+    );
```
Comment (skarimo) [inline src/external-actor-on-team-approved-pr.ts:76]: Nit: (feel free to ignore) Would be nice to use the logging provided in the core package: https://github.com/actions/toolkit/tree/main/packages/core#logging

**Comment by jhgilbert** [comment]
@skarimo, thanks for the review! It's ready for another look.

**Review by skarimo — APPROVED** [review]

**Pull request merged**